crud_for_code_review.py

```diff
@@ -0,0 +1,546 @@
+                image_url=image_url)
+
+    if User.query.get(user_id):
+        print("User already exists")
+    else:
+        db.session.add(user)
+        db.session.commit()
+
+    return user
+
+
+def create_user_artist(user_id, artist_id, api_type):
+    """Create and return a new UserArtist"""
+
+    user_artist = UserArtist(user_id=user_id, artist_id=artist_id, api_type=api_type)
+
+    db.session.add(user_artist)
+    db.session.commit()
+
+    return user_artist
+
+
+def create_artist(artist_id, artist_name, popularity, image_url):
+    """Create and return a new Artist"""
+
+    artist = Artist(artist_id=artist_id, 
+                    artist_name=artist_name, 
+                    popularity=popularity,
+                    image_url=image_url)
+
+    db.session.add(artist)
+    db.session.commit()
+
+    return artist
+
+
+def create_genre(genre):
+    genre = Genre(genre=genre)
+
+    db.session.add(genre)
+    db.session.commit()
+
+    return genre
+
+
+def create_artist_genres(artist_id, genre_id):
+
+    artist_genres = ArtistGenre(artist_id=artist_id,
+                                genre_id=genre_id)
+
+    db.session.add(artist_genres)
+    db.session.commit()
+
+    return artist_genres
+
+#query helper functions
+def get_artist_by_id(artist_id):
+    """Query artists table and return object"""
+
+    return Artist.query.get(artist_id)
+
+
+def check_user_artists(user_id, artist_id, api_type):
+    """Check if an artist is in a user's artists"""
+
+    return db.session.query(UserArtist).filter((UserArtist.user_id==user_id), (UserArtist.artist_id==artist_id), (UserArtist.api_type==api_type)).first()
+
+
+def check_artist_genres(artist_id, genre_id):
+    """Check if a genre is in an artist's genres"""
+
+    return ArtistGenre.query.filter_by(artist_id=artist_id).first() and ArtistGenre.query.filter_by(genre_id=genre_id).first()    
+
+
+def get_genre_by_name(genre):
+
+    return Genre.query.filter_by(genre=genre).first()
+
+
+def get_genre_id_by_name(genre):
+
+    return db.session.query(Genre.genre_id).filter_by(genre=genre).first()
+
+
+def get_genres_by_user_artists(user_id):
+    """returns {'art pop': ['Grimes', 'Lady Lamb'], 
+    'electropop': ['Grimes', 'Big Wild'], 
+    'indietronica': ['Grimes', 'Big Wild'],..}
+    includes repeating artists in genres"""
+
+    #join tables to access relationships
+    #user_join returns a User object
+    user_join = User.query.options( 
+             db.joinedload('artists') # attribute for user 
+               .joinedload('genres')  # attribute from artist
+            ).get(user_id)  
+
+    user_genres = {}                                                                                                                 
+
+    for artist in user_join.artists: 
+        for genre in artist.genres: 
+             user_genres[genre.genre] = user_genres.get(genre.genre, []) + [artist.artist_name]
+
+    return user_genres
+
+
+def count_user_artists_by_genre(user_id):
```

Comment:
I'm a bit confused about what this function is doing. Since `user_genres` is a dict where the keys are the genre name, the same genre wont ever appear twice in line 130, right? So why do you first check `count_artists.get(genre, 0)`? It looks like `get_genres_by_user_artists` is only called by this function, so in `get_genres_by_user_artists`, why not just increment the count of artists instead of appending all of the artist names?